### PR TITLE
Add fusion testing code for fullOp and expandOp

### PR DESCRIPTION
**Pull request by gongshaotian**
### PR types

New features
### PR changes

OPs
### Description

1. Add fusion testing code for full and expand
2. Add support for pd.full in ir op creator
3. Add support for Place attribute
4. Add vector<int64 in IrAttrTypeCast_ t> Property support for types

### CODE REVIEW

File: paddle/fluid/ir/drr/match_context_impl.h [hunk 1]
```diff
@@ -62,6 +63,28 @@ struct IrAttrTypeCast<std::vector<int32_t>> {
...
+    auto array_attr = attr.dyn_cast<ir::ArrayAttribute>();   
+
+    if (array_attr) {
+      for(size_t i = 0; i < array_attr.size(); i++){
+        result.push_back(array_attr.at(i).dyn_cast<ir::Int64Attribute>().data());
+      }
+      return result;
+    } 
+
+    else if (attr.dyn_cast<paddle::dialect::IntArrayAttribute>()) {
```
Comment (zyfncg) [inline paddle/fluid/ir/drr/match_context_impl.h]: else 直接加到 } 后面

File: paddle/fluid/ir/drr/match_context_impl.h [hunk 2]
```diff
@@ -62,6 +63,28 @@ struct IrAttrTypeCast<std::vector<int32_t>> {
...
+      result = attr.dyn_cast<paddle::dialect::IntArrayAttribute>().data().GetData();
+      return result;
+    }
+    LOG(ERROR) << "Dynamic cast failed for IR attribute vector<int64_t>";
+    IR_THROW();
```
Comment (zyfncg) [inline paddle/fluid/ir/drr/match_context_impl.h]: 错误信息直接合并到IR_THROW里面

File: test/cpp/ir/pattern_rewrite/drr_test.cc [hunk 3]
```diff
@@ -45,31 +45,37 @@ class RemoveRedundentReshapePattern
   }
 };
 
-class FoldBroadcastToConstantPattern
-    : public ir::drr::DrrPatternBase<FoldBroadcastToConstantPattern> {
+class RemoveExpandPattern
+    : public ir::drr::DrrPatternBase<RemoveExpandPattern> {
```
Comment (zyfncg) [inline test/cpp/ir/pattern_rewrite/drr_test.cc]: 这也算是常量折叠的一种case, 或者可以叫 `FoldExpandToConstant`

File: paddle/fluid/ir/drr/match_context_impl.h (hunk 1 above)
Comment (gongshaotian) [reply paddle/fluid/ir/drr/match_context_impl.h]: 已修改

File: paddle/fluid/ir/drr/match_context_impl.h (hunk 2 above)
Comment (gongshaotian) [reply paddle/fluid/ir/drr/match_context_impl.h]: 已修改

File: test/cpp/ir/pattern_rewrite/drr_test.cc (hunk 3 above)
Comment (gongshaotian) [reply test/cpp/ir/pattern_rewrite/drr_test.cc]: 已修改

**Comment by gongshaotian** [comment]
完成修改

**Pull request merged**